information/specification.md

```diff
@@ -0,0 +1,706 @@
+* Not all **properties** of coin selection algorithms are included.
+* There are currently no **worked examples** or **test vectors**.
+
+# Contents
+
+* [Purpose](#purpose)
+* [Background](#background)
+  * [What is Coin Selection?](#what-is-coin-selection)
+  * [Why is Coin Selection Non-Trivial?](#why-is-coin-selection-non-trivial)
+    * [Issues](#issues)
+    * [Goals](#goals)
+* [Definitions](#definitions)
+  * [Address](#address)
+  * [Coin Amount](#coin-amount)
+  * [Transaction Input](#transaction-input)
+  * [Transaction Output](#transaction-output)
+  * [UTxO Set](#utxo-set)
+* [Interface](#interface)
+  * [Parameters](#parameters)
+    * [Requested Output List](#requested-output-list)
+    * [Initial UTxO Set](#initial-utxo-set)
+    * [Maximum Input Count](#maximum-input-count)
+  * [Results](#results)
+    * [Coin Selection](#coin-selection)
+    * [Remaining UTxO Set](#remaining-utxo-set)
+  * [Properties](#properties)
+    * [Coverage of Payments](#coverage-of-payments)
+    * [Correctness of Change](#correctness-of-change)
+    * [Conservation of UTxO](#conservation-of-utxo)
+    * [Conservation of Outputs](#conservation-of-outputs)
+    * [Cardinality](#cardinality)
+  * [Failure Modes](#failure-modes)
+    * [UTxO Balance Insufficient](#utxo-balance-insufficient)
+    * [UTxO Not Fragmented Enough](#utxo-not-fragmented-enough)
+    * [UTxO Fully Depleted](#utxo-fully-depleted)
+    * [Maximum Input Count Exceeded](#maximum-input-count-exceeded)
+* [Implementations](#implementations)
+  * [Largest-First](#largest-first)
+    * [State](#state)
+      * [Available UTxO List](#available-utxo-list)
+      * [Unpaid Output List](#unpaid-output-list)
+      * [Accumulated Coin Selection](#accumulated-coin-selection)
+    * [Computation](#computation)
+    * [Termination](#termination)
+  * [Random-Improve](#random-improve)
+    * [Motivating Principles](#motivating-principles)
+      * [Principle 1: Dust Management](#principle-1-dust-management)
+      * [Principle 2: Change Management](#principle-2-change-management)
+      * [Principle 3: Performance Management](#principle-3-performance-management)
+    * [State](#state-1)
+      * [Available UTxO Set](#available-utxo-set)
+      * [Accumulated Coin Selection](#accumulated-coin-selection-1)
+    * [Computation](#computation-1)
+      * [Phase 1: Random Selection](#phase-1-random-selection)
+      * [Phase 2: Improvement](#phase-2-improvement)
+    * [Termination](#termination-1)
+* [External Resources](#external-resources)
+  * [Self Organization in Coin Selection](#self-organization-in-coin-selection)
+
+# Purpose
+
+The purpose of this article is to describe, in _human-readable terms_, the coin
+selection algorithms used by Cardano Wallet and other parts of the Cardano
+ecosystem.
+
+# Background
+
+## What is Coin Selection?
+
+Coin selection is the process of choosing unspent coins from a wallet in order
+to pay money to one or more recipients.
```

Comment:
Shouldn't the document actually starts with the definitions? For example, here we are already talking about _coins_ and _wallet_ without having really defined those terms. Unless the sense given to these terms is the common sense of a coin and wallet.

Reply:
> Shouldn't the document actually starts with the definitions? 

That's a good question.

My original intention was to have a brief, human-readable introduction that _doesn't_ rely on the reader to first have to read through a set of definitions.

The question I'm now considering is this: would a reader that is representative of our target audience be able to comprehend the introduction section _without_ too much difficulty, and _without_ having to jump to the definitions section?

One possible alternative solution: we could add more **hyperlinks** from terms to their definitions, as we have done in other places. Thoughts?

Reply:
@KtorZ I've now significantly revised both the background and definitions sections.

I've tried to keep the background section self-contained, so that it is readable by someone who hasn't necessarily read the definitions.

Though where appropriate, I've added hyperlinks from the background to the definitions, so the reader can gain greater understanding when they feel it is necessary.

Feel free to let me know if you would like to see further revisions!